operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/kubernetes/KubernetesDependentResourceInitializer.java

```diff
@@ -0,0 +1,68 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent.kubernetes;
+
+import java.lang.reflect.InvocationTargetException;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.config.dependent.KubernetesDependentResourceConfiguration;
+import io.javaoperatorsdk.operator.api.config.informer.InformerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResourceInitializer;
+
+import static io.javaoperatorsdk.operator.api.config.Utils.valueOrDefault;
+
+public class KubernetesDependentResourceInitializer<T extends DependentResource<?, ?>>
+    implements DependentResourceInitializer<T> {
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public T initialize(
+      Class<T> dependentResourceClass,
+      ControllerConfiguration<?> configurationService,
```

Comment:
This shouldn't be named `configurationService` as it is confusing.

Reply:
will fix